v0/destinations/delighted/util.js

```diff
@@ -0,0 +1,85 @@
+const axios = require("axios");
+const { CustomError, getValueFromMessage } = require("../../util");
+const { ENDPOINT } = require("./config");
+
+const isValidEmail = email => {
+  const re = /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
+  return re.test(String(email).toLowerCase());
+};
+const isValidPhone = phone => {
+  const phoneformat = /^\+[1-9]\d{10,14}$/;
+  return phoneformat.test(String(phone));
+};
+
+const isValidUserIdOrError = (channel, userId) => {
+  if (channel === "email") {
+    if (!isValidEmail(userId)) {
+      throw new CustomError("Email format is not correct.", 400);
```

Comment:
can we change the error message to `Email format in userId is not correct.` ? same for the phone.
and in the final else block, would it be more appropriate to give the message as "invalid channel type".
wdyt?